### ISSUE TITLE
Unclutter `epi_archive`'s `print` method + add preview of `DT`

### ISSUE THREAD

**Issue by brookslogan** [cmu-delphi/epiprocess#341]
Pretty commonly, we'll print an `epi_archive` `x`, see a lot but not very useful, then print `x$DT` to get a better idea what things look like.  We should rebalance the `print` output to include the `DT`, exclude any redundant information, and move some other information into a `summary` method.  Then we might be able to make `DT` a private field without much issue, and avoid some worries about aliasing and mutation and `key` invalidation.

**Comment by brookslogan** [comment]
Another note from @dsweber2 : we should consider `head` and `tail` methods.  Probably after #340.

**Comment by brookslogan** [comment]
> Then we might be able to make DT a private field without much issue, and avoid some worries about aliasing and mutation and key invalidation.

Making DT a private field is out of scope.  This issue is a blocker to it, though (as is implementing `filter` for `epi_archive`s).

**Comment by brookslogan** [comment]
> Another note from @dsweber2 : we should consider head and tail methods. Probably after https://github.com/cmu-delphi/epiprocess/issues/340.

This is out of scope.
